invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/AffiliationsField/AffiliationsField.js

```diff
@@ -54,7 +54,7 @@ export class AffiliationsField extends Component {
                 );
               }}
               value={getIn(values, fieldPath, []).map(
-                (val) => val.name || val.text || val.id
+                (val) => val.id || val.name || val.text
```

Comment:
Looks good to me, just wondering if it would be enough simply the `id`, not sure why we keep the name and text as fallback